app/main.py

```diff
@@ -2,0 +1,23 @@
+from app.cafe import Cafe
+
+from app.errors import (
+    NotWearingMaskError,
+    VaccineError
+)
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    masks_to_buy = 0
+
+    for person in friends:
+        try:
+            cafe.visit_cafe(person)
+        except VaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            masks_to_buy += 1
+
+    if masks_to_buy:
+        return f"Friends should buy {masks_to_buy} masks"
```

Comment:
Code Efficiency Issue: You have done a good job here comparing `masks_to_buy` to 0. However, it is not necessary to compare a variable to `0` if there is no need. You can simplify this code by doing: 

```python
if masks_to_buy:
    return f"Friends should buy {masks_to_buy} masks"
```